psse/psse-converter/src/main/java/com/powsybl/psse/converter/PsseImporter.java

```diff
@@ -182,30 +211,99 @@ private static void createLoad(PsseLoad psseLoad, ContainersMapping containerMap
-        String busId = getBusId(psseShunt.getI());
-        VoltageLevel voltageLevel = network.getVoltageLevel(containerMapping.getVoltageLevelId(psseShunt.getI()));
-        ShuntCompensatorAdder adder = voltageLevel.newShuntCompensator()
-                .setId(busId + "-SH" + psseShunt.getId())
-                .setConnectableBus(busId)
-                .setBus(busId)
-                .setSectionCount(1);
-        adder.newLinearModel()
-                .setBPerSection(psseShunt.getBl())//TODO: take into account gl
-                .setMaximumSectionCount(1)
-                .add();
-        ShuntCompensator shunt = adder.add();
+    private static void createShuntCompensator(PsseFixedShunt psseShunt, ContainersMapping containerMapping, Network network) {
+        if (psseShunt.getBl() != 0) {
+            String busId = getBusId(psseShunt.getI());
+            VoltageLevel voltageLevel = network.getVoltageLevel(containerMapping.getVoltageLevelId(psseShunt.getI()));
+            ShuntCompensatorAdder adder = voltageLevel.newShuntCompensator()
+                    .setId(busId + "-SH" + psseShunt.getId())
+                    .setConnectableBus(busId)
+                    .setBus(busId)
+                    .setSectionCount(1);
+            adder.newLinearModel()
+                    .setBPerSection(psseShunt.getBl())//TODO: take into account gl
+                    .setMaximumSectionCount(1)
+                    .add();
+            ShuntCompensator shunt = adder.add();
+
+            if (psseShunt.getStatus() == 1) {
+                shunt.getTerminal().connect();
+            }
 
-        if (psseShunt.getStatus() == 1) {
-            shunt.getTerminal().connect();
+            if (psseShunt.getGl() != 0) {
+                LOGGER.warn("Shunt Gl not supported ({})", psseShunt.getI());
+            }
+        } else {
+            LOGGER.warn("Shunt ({}) has Bl = 0, not imported ", psseShunt.getI()); //TODO : allow import of shunts with Bl= 0 in iidm?
+        }
+    }
+
+    private void createSwitchedShuntBlocMap(PsseRawModel psseModel, HashMap<PsseSwitchedShunt, ShuntBlockTab > stoBlockiTab) {
+
+        /* Creates a map between the PSSE switched shunt and the blocks info of this shunt
+        A switched shunt may contain up to 8 blocks and each block may contain up to 9 steps of the same value (in MVAR)
+        A block may be capacitive or inductive */
+        for (PsseSwitchedShunt psseSwShunt : psseModel.getSwitchedShunts()) {
+
+            ShuntBlockTab sbt = new ShuntBlockTab();
+
+            int[] ni = new int[8];
+            ni[0] = psseSwShunt.getN1();
+            ni[1] = psseSwShunt.getN2();
+            ni[2] = psseSwShunt.getN3();
+            ni[3] = psseSwShunt.getN4();
+            ni[4] = psseSwShunt.getN5();
+            ni[5] = psseSwShunt.getN6();
+            ni[6] = psseSwShunt.getN7();
+            ni[7] = psseSwShunt.getN8();
```

Comment:
I think it could be refactor. Maybe we could introduce a method in psseSwShunt to get this array if it makes sense. Note that it's possible to write
```
int[] ni = {psseSwShunt.getN1(), psseSwShunt.getN2().... };
```

Reply:
OK, proposition of modification done